### PR TITLE
Remove catalog implementation

### PR DESCRIPTION
**Pull request by Tansito**
### Summary

This PR reset changes from #1181, #1204, #1211

### CODE REVIEW

**Comment by psschwei** [comment]
nit: since this is deleting a bunch of code, I'd retitle to something like "Remove catalog implementation"

**Comment by Tansito** [comment]
How these migrations where not published I think I preferred to remove them but let me know if you prefer to create a new migration file. I would agree too.

**Comment by akihikokuroda** [comment]
>How these migrations where not published I think I preferred to remove them but let me know if you prefer to create a new migration file. I would agree too.

What number is used for the next migration file?  Is it the next to the existing highest one or one of the removed ones? @Tansito

**Comment by Tansito** [comment]
I think it got the last one but not 100% sure. Let me confirm it, @akihikokuroda , good question

File: gateway/api/migrations/0014_runtimejob.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 class Migration(migrations.Migration):
 
     dependencies = [
-        ("api", "0013_program_groups_catalogentry"),
+        ("api", "0012_jobconfig_python_version"),
```
Comment (IceKhan13) [inline gateway/api/migrations/0014_runtimejob.py:10]: Removing migrations or changing order of migration will break existing databases. If we want to remove code, related to models we need to remove models and generate new migration which will delete tables from datase.

Long story short: migrations are always append only :)

**Review by IceKhan13 — CHANGES_REQUESTED** [review]

File: gateway/api/migrations/0014_runtimejob.py (hunk 1 above)
Comment (IceKhan13) [reply gateway/api/migrations/0014_runtimejob.py:10]: oops! Just saw your last comment about not published. 

I think it is still better to append migrations, if it is not hard to do

Comment (Tansito) [reply gateway/api/migrations/0014_runtimejob.py]: Solved in [674a79c](https://github.com/Qiskit-Extensions/quantum-serverless/pull/1276/commits/674a79c9350452c8bc97fef38866a5c8f933dc3b)

**Review by IceKhan13 — APPROVED** [review]
Thank you @Tansito !

**Review by akihikokuroda — APPROVED** [review]

**Pull request merged**